src/test/java/com/photolocos/enterprise/EnterpriseApplicationTests.java

```diff
@@ -2,62 +2,92 @@
 
 
     @Test
     void contextLoads() {
     }
 
+    @Test
+    void userSearchesLocation_returnsPhotoResults() throws Exception {
+        givenPhotoWithStateLocationExists();
+        returnsPhotoResults(userSearchesLocation());
+    }
+
+    private void givenPhotoWithStateLocationExists() throws Exception {
+        Photo photo = new Photo();
+        Location location = new Location();
+        location.setState("Ohio");
+
+        photo.setLocation(location);
+        photoDAO.createEntry(photo);
+    }
+
+    private Set<Photo> userSearchesLocation() {
+        return photoDAO.fetchByLocation("Ohio");
+    }
+
+    private void returnsPhotoResults(Set<Photo> results) {
+        Assert.notNull(results, "Photo search results returned null");
+        Assert.notEmpty(results, "Photo search results returned empty set.");
+
+        boolean locationMatches = true;
+        for(Photo photo: results){
+            if(!photo.getLocation().getState().equals("Ohio")){
+                locationMatches = false;
+                break;
+            }
+        }
+
+        Assert.isTrue(locationMatches, "Photo search returned an incorrect result set.");
+    }
+
     @Test
     void fetchPhotoByTag_returnsRiverForTagRiver() throws Exception {
-        givenPhotoDataAreAvailable();
         whenPhotoAddedWithTagRiver();
-        whenSearchPhotoWithTagRiver();
-        thenReturnPhotosWithTagRiver();
+        thenReturnPhotosWithTagRiver(whenSearchPhotoWithTagRiver());
     }
 
-    private void whenPhotoAddedWithTagRiver() {
-        String[] tags = {"river"};
-        photo.setTags(tags);
-        Mockito.when(photoDAO.fetchByTag(tags)).thenReturn((Set<PhotoDTO>) photo);
-    }
+    private void whenPhotoAddedWithTagRiver() throws Exception {
+        Photo photo = new Photo();
+        photo.setTags(new String[]{"river"});
 
-    private void givenPhotoDataAreAvailable() {
-        //Mockito.when(PhotoDAO.save(photo)).thenReturn(photo);
-        /**
-         * Save method not yet created in PhotoDAO
-         * */
-        photoDAO.save(photo);
+        photoDAO.createEntry(photo);
     }
 
-    private void whenSearchPhotoWithTagRiver() {
-        String[] tag = {"river"};
-        photo = (PhotoDTO) photoDAO.fetchByTag(tag);
+    private Set<Photo> whenSearchPhotoWithTagRiver() {
+        return photoDAO.fetchByTag(new String[]{"river"});
     }
 
-    private PhotoDTO thenReturnPhotosWithTagRiver() {
-        return photo;
+    private void thenReturnPhotosWithTagRiver(Set<Photo> results) {
+        Assert.notNull(results, "Photo search results returned null");
+        Assert.notEmpty(results, "Photo search results returned empty set.");
+
+        boolean photosMatch = true;
+        for(Photo photo: results){
+            if(!Arrays.asList(photo.getTags()).contains("river")){
+                photosMatch = false;
+            }
+        }
+
+        Assert.isTrue(photosMatch, "Photo search returned an incorrect result set.");
```

Comment:
Here I fixed your existing unit test which wasn't passing and caused the project to fail to compile. [(SonarQube Seven Deadly Sins: Bugs and Potential Bugs)](https://docs.sonarqube.org/display/SONARQUBE45/Bugs+and+Potential+Bugs)